File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.java
import java.util.List;
import java.util.ArrayList;

class Solution {
    public List<Integer> solution(int[] arr, boolean[] flag) {
        List<Integer> answer = new ArrayList<>();
        
        for(int i = 0; i < flag.length; i++){
            if(flag[i]){
                for(int j = 0; j < arr[i]*2; j++){
                    answer.add(arr[i]);
                }
            }
            else{
                for(int j = 0; j < arr[i]; j++){
                    answer.remove(answer.size() - 1);
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.py
def solution(arr, flag):
    answer = []
    for i in range(len(flag)):
        if(flag[i]):
            answer.extend([arr[i] for _ in range(arr[i]*2)])
        else:
            [answer.pop() for _ in range(arr[i])]
    
    return answer